crates/ide_assists/src/handlers/replace_derive_with_manual_impl.rs

```diff
@@ -240,8 +333,128 @@ struct Foo {
 }
 
 impl fmt::Debug for Foo {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        ${0:todo!()}
+    $0fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("Foo").field("bar", &self.bar).finish()
+    }
+}
+"#,
+        )
+    }
+    #[test]
+    fn add_custom_impl_debug_tuple_struct() {
+        check_assist(
+            replace_derive_with_manual_impl,
+            r#"
+mod fmt {
+    pub struct Error;
+    pub type Result = Result<(), Error>;
+    pub struct Formatter<'a>;
+    pub trait Debug {
+        fn fmt(&self, f: &mut Formatter<'_>) -> Result;
+    }
+}
```

Comment:
Would be great to move this to minicore(`crates/test_utils/src/minicore.rs`) I believe.

Reply:
@Veykril I'd be happy to, but I'm not familiar with how minicore works. Can you provide more detailed instructions on what you want me to do here?

Reply:
Basically add a `fmt` flag to the doc string in the minicore file under the `Available flags:` line and then add the corresponding snippet to the contents, that is:

```rs
// region:fmt
pub mod fmt {
    pub struct Error;
    pub type Result = Result<(), Error>;
    pub struct Formatter<'a>;
    pub trait Debug {
        fn fmt(&self, f: &mut Formatter<'_>) -> Result;
    }
}
// endregion:fmt
```

Reply:
In short, minicore is a stripped down version of the core lib where we can enable parts of it on demand for tests, that way we can still benefit from core dependent things in tests without having to load the entire core crate.

Usage wise for your tests you can then remove your `mod fmt {...}` stuff and instead add a `//- minicore:fmt` line at the top of the fixtures.

Reply:
So with this test here as an example, it would then become:
```rs
    #[test]
    fn add_custom_impl_debug_tuple_struct() {
        check_assist(
            replace_derive_with_manual_impl,
            r#"
//- minicore:fmt
#[derive(Debu$0g)]
struct Foo(String, usize);
"#,
            r#"
struct Foo(String, usize);
impl fmt::Debug for Foo {
    $0fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        f.debug_tuple("Foo").field(&self.0).field(&self.1).finish()
    }
}
"#
    );
}
```

Reply:
Oh I see! - that's neat!

Updated the tests to use minicore!